### PR TITLE
gmtspatial -N+i+pstart is allowed

### PR DESCRIPTION
**Pull request by PaulWessel**
The updated **gmtspatial** checks would refuse **+p**_start_ if **+i** is used.  Now that is relaxed, the usage messages are cleaned up a bit, as is the docs.

### CODE REVIEW

**Review by joa-quim — COMMENTED** [review]
Hmm, I was working on this too. It's not only +i+p, it's also the bad results.
I'll submit my PR too for you to look.

**Comment by joa-quim** [comment]
OK, back to this. Just allowing **+p** does not make it work
```
C:\v>gmt spatial cloud.txt -Npols.txt+i+p1
141     15      0
142     15      0
143     15      0
141     18      0
142     18      0
138     15      1
139     15      1
138     12      1
139     12      1
138     15      1
```

**Comment by PaulWessel** [comment]
Update and try again.

File: src/gmtspatial.c [hunk 1]
```diff
@@ -2029,12 +2039,13 @@ EXTERN_MSC int GMT_gmtspatial (void *V_API, int mode, void *args) {
...
 		T = C->table[0];	/* Only one input file so only one table */
 		count = gmt_M_memory (GMT, NULL, D->n_segments, unsigned int);
+		ID = Ctrl->N.ID - 1;	/* May be overridden below by header info, otherwise will be incremented to 0 (or whatever +p set) */
 		for (seg2 = 0; seg2 < T->n_segments; seg2++) {	/* For all polygons */
 			S2 = T->segment[seg2];
 			SH = gmt_get_DS_hidden (S2);
 			if (gmt_polygon_is_hole (GMT, S2)) continue;	/* Holes are handled in gmt_inonout */
 			GMT_Report (API, GMT_MSG_INFORMATION, "Look for points/features inside polygon segment %" PRIu64 " :\n", seg2);
-			if (Ctrl->N.ID == 0) {	/* Look for polygon IDs in the data headers */
+			if (!Ctrl->N.p_set) {	/* Look for polygon IDs in the data headers */
```
Comment (joa-quim) [inline src/gmtspatial.c:2048]: But if you do this -Z and -L won't be searched in headers unless +p is used.

Comment (joa-quim) [reply src/gmtspatial.c:2048]: See what I did in #7749

Comment (PaulWessel) [reply src/gmtspatial.c:2048]: Dont get it. If p is not used then Ctrl->N.p_set is false and !false is true and it goes to look in the headers. No?

Comment (PaulWessel) [reply src/gmtspatial.c:2048]: Looks like we are yelling at the same time!

**Comment by PaulWessel** [comment]
IF works OK for you then pls approve.

**Review by joa-quim — APPROVED** [review]
What a mess with these 2 PRs. I approved my own one.

**Pull request merged**